GLMakie/src/GLAbstraction/GLRender.jl

```diff
@@ -96,75 +96,99 @@ function render(renderobject::RenderObject, vertexarray=renderobject.vertexarray
     return
 end
 
+function vao_boundscheck(target::Integer, current::Integer, vao)
+    if target <= current # assuming 0-based OpenGL indices
+        msg = IOBuffer()
+        print(msg, "BoundsError: OpenGL vertex index $current exceeds the number of vertices $target.\n Occurred with ")
+        show(msg, MIME"text/plain"(), vao)
+        error(String(take!(msg)))
+    end
+end
 
+# multiple index ranges
 """
-Renders a vertexarray, which consists of the usual buffers plus a vector of
-unitranges which defines the segments of the buffers to be rendered
+    render(vao::GLVertexArray[, mode = GL_TRIANGLES])
+
+Renders a vertexarray based on its `vao.indices` type.
 """
 function render(vao::GLVertexArray{T}, mode::GLenum=GL_TRIANGLES) where T <: VecOrSignal{UnitRange{Int}}
+    N_vert = length(vao)
     for elem in to_value(vao.indices)
+        # TODO: Should this exclude last(elem), i.e. shift a:b to (a-1):(b-1)
+        #       instead of (a-1):b?
+        vao_boundscheck(N_vert, last(elem), vao)
         glDrawArrays(mode, max(first(elem) - 1, 0), length(elem) + 1)
```

Comment:
I basically just want to highlight the TODO I added here. I think this should just shift indices from 1-based to 0-based but with length(elem) + 1 that's not what's happening. Is this a mistake?